core/src/test/scala/unit/kafka/server/AbstractCreateTopicsRequestTest.scala

```diff
@@ -136,11 +144,13 @@ abstract class AbstractCreateTopicsRequestTest extends BaseRequestTest {
         }
       }
 
-      // Verify controller broker has the correct metadata
-      verifyMetadata(controllerSocketServer)
+      if (!isKRaftTest()) {
```

Comment:
Why not verify metadata in Kraft mode?

Reply:
In ZK mode, the ZK change is done by the specific broker that receives the CreateTopicsRequest.  Therefore, we know that that broker's metadata cache will get updated immediately. This isn't the case in KRaft mode. In KRaft mode the change is made by the controller, not the broker, so the broker's metadata cache will not be immediately updated. Therefore, we have to wait for metadata propagation, which we do on the next few lines. So, we are verifying that metadata, just after waiting for the propagation delay.

This is something that actually will change even in ZK mode, once we have KIP-590 enabled. We probably will just have to get rid of this check even in ZK mode once we enable that.